src/Spice86.Core/Emulator/Function/ExecutionFlowRecorder.cs

```diff
@@ -230,4 +236,32 @@ public class ExecutionFlowRecorder {
         }
         destinationAddresses.Add(new SegmentedAddress(toCS, toIP));
     }
+
+    public string Dump() {
+        return _callStack.Dump();
+    }
+}
+
+public class CircularStringBuffer {
```

Comment:
There already is a byte array based CircularBuffer, in the Spice86.Core.Emulator.Devices.Sound.Blaster namespace, perhpas the two classes could be merge as a single class, using generics ? Or a base class ?

Reply:
I'm not sure they are that comparable really.  The one deals with ranges of bytes while the other with singular items. I did move it to it;s own file and improved comments.